File: src/backend/laravel-app/app/Models/ReglementInterieur.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

/**
 * @OA\Schema(
 *     required={"libelle"},
 *     @OA\Xml(name="ReglementInterieur"),
 *     @OA\Property(property="id", type="integer", readOnly=true, example="1"),
 *     @OA\Property(property="libelle", type="string", readOnly=true, description="Un nouveau reglement interieur"),
 *     @OA\Property(property="created_at", ref="#/components/schemas/BaseModel/properties/created_at"),
 *     @OA\Property(property="updated_at", ref="#/components/schemas/BaseModel/properties/updated_at"),
 *     @OA\Property(property="deleted_at", ref="#/components/schemas/BaseModel/properties/deleted_at")
 * )
 *
 * @method static create(array $array)
 * @method static findOrFail($id)
 * @method static paginate(int $int)
 *
 * Class ReglementInterieur
 *
 * @package App\Models
 */
class ReglementInterieur extends Model
{
    use HasFactory;

    protected $fillable = [
        'libelle'
    ];
    public function regle(){
        return $this->belongsToMany(Regle::class, 'reglementInterieurId');
    }
}

File: src/frontend/mobile/flutter_app/lib/common/models/reglement_interieur.dart
class ReglementInterieur {
  final int id;
  final String libelle;
  final String createdAt;
  final String updatedAt;

  ReglementInterieur({
    required this.id,
    required this.libelle,
    required this.createdAt,
    required this.updatedAt,
  });

  factory ReglementInterieur.fromJson(Map<String, dynamic> json) =>
      ReglementInterieur(
          id: json['id'],
          libelle: json['libelle'],
          createdAt: json['created_at'],
          updatedAt: json['updated_at']);
}
